compiler/rustc_mir/src/interpret/operand.rs

```diff
@@ -558,7 +556,9 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         layout: Option<TyAndLayout<'tcx>>,
     ) -> InterpResult<'tcx, OpTy<'tcx, M::PointerTag>> {
         match val.val {
-            ty::ConstKind::Param(_) | ty::ConstKind::Bound(..) => throw_inval!(TooGeneric),
+            ty::ConstKind::Param(_) |
+            // FIXME(oli-obk): is this unreachable?
+            ty::ConstKind::Bound(..) => throw_inval!(TooGeneric),
```

Comment:
i think this should be reachable :thinking: but afaik know only with `feature(const_evaluatable_checked)` so it might be good to change this to a `bug!` until we find a test case.